configure

```diff
@@ -918,7 +882,7 @@ EOF
       echo "           Fortran compiler is $SFC_arch"
       echo "              It will build in $netcdf_arch"
       echo " "
-      if [ -e $NETCDF/bin/nc-config ] ; then
+      if [ -e $NETCDF_C/bin/nc-config ] ; then
         echo "NetCDF version: ${ncversion}"
         echo "Enabled NetCDF-4/HDF-5: `nc-config --has-nc4`"
```

Comment:
I believe this will output the results from the wrong `nc-config` if `$NETCDF_C` differs from default in `PATH`

Reply:
Most of these changes resolve #1742 and having consistent usage of `nf-config` and `nc-config` fall under the aim of this PR

Pinging @wkliao to be aware that this should be the PR that resolves issues noted in #1742

Reply:
You're right, I missed that one. I will push the appropriate change in a few minutes.

Reply:
> Most of these changes resolve #1742 and having consistent usage of `nf-config` and `nc-config` fall under the aim of this PR
> 
> Pinging @wkliao to be aware that this should be the PR that resolves issues noted in #1742

If this PR #1923 resolved issues raised by PR #1742 and beyond, then #1742 can be closed. Given #1742 was created 1 and 1/2 years ago, I wonder if there is anything it missed, so it could not be merged.

Reply:
I think the primary thing #1742 did not have was the ability to use both `nf-config` and `nc-config` for different library link info. This probably wasn't a high priority then, but netCDF's shift to split the two programs even further as of v4.9.2 will soon be problematic for WRF's current make builds. (see Unidata/netcdf-c#2619 and [Unidata/netcdf-c v4.9.2 release notes](https://github.com/Unidata/netcdf-c/releases/tag/v4.9.2))